drv/ext-int-ctrl-api/src/lib.rs

```diff
@@ -9,7 +9,7 @@ use userlib::*;
 
 #[derive(Copy, Clone, Debug, FromPrimitive, IdolError)]
 pub enum ExtIntCtrlError {
-    IRQUnassigned,
+    IRQUnassigned = 1,
```

Comment:
Is there a way to add a compile time check to ensure the error codes are always non-zero?

Reply:
Not really, Rust unfortunately doesn't have any form of static assert yet. We could add something to the `build.rs` that would check this, but that would still require new idol interfaces to include that `build.rs`. Maybe there is some way to add this check to [idolatry](https://github.com/oxidecomputer/idolatry), but I'm not totally sure of how we'd do it.

Reply:
We could also use something like [this](https://docs.rs/static_assertions/0.1.1/static_assertions/index.html).  It seems to be the standard way of doing compile time checks right now.

Reply:
Ack. I think we can follow this up on a separate JIRA issue so that we don't block this PR.

Reply:
Made a [ticket](https://rivosinc.atlassian.net/browse/SW-599)